backend/src/friends/friends.service.ts

```diff
@@ -0,0 +1,92 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { User } from '../users/entities/user.entity';
+import { UsersService } from '../users/users.service';
+import { FRIEND_STATUS, Friend } from './entities/friend.entity';
+
+@Injectable()
+export class FriendsService {
+  constructor(
+    @InjectRepository(Friend)
+    private readonly friendRepository: Repository<Friend>,
+    private readonly usersService: UsersService,
+  ) {}
+
+  // TODO: Split into received and sent requests
+  async listRequests(userId: number): Promise<Friend[]> {
+    return await this.friendRepository.find({
+      where: [
+        { sender: { id: userId }, status: FRIEND_STATUS.PENDING },
+        { recipient: { id: userId }, status: FRIEND_STATUS.PENDING },
+      ],
+      relations: ['sender', 'recipient'],
+    });
+  }
+
+  async listFriends(userId: number): Promise<User[]> {
+    const acceptedFriendRequests = await this.friendRepository.find({
+      where: [
+        { sender: { id: userId }, status: FRIEND_STATUS.ACCEPTED },
+        { recipient: { id: userId }, status: FRIEND_STATUS.ACCEPTED },
```

Comment:
do we want to list BLOCKED friends as well? So maybe we can unblock them if we want. And if we want to unblock them we need a new route, right?

Reply:
Tbh I don't know about this one, because I'm not entirely sure we're going to block users from other users or just users from chats. If the later, I think we may just drop this additional route